pkg/istio-agent/xds_proxy.go

```diff
@@ -412,6 +413,14 @@ func (p *XdsProxy) getCertKeyPaths(agent *Agent) (string, string) {
 	if agent.secOpts.ProvCert != "" {
 		key = path.Join(agent.secOpts.ProvCert, constants.KeyFilename)
 		cert = path.Join(path.Join(agent.secOpts.ProvCert, constants.CertChainFilename))
+
+		// CSR may not have completed – use JWT to auth.
+		if _, err := os.Stat(key); os.IsNotExist(err) {
```

Comment:
@costinm @williamaronli are there likely to be issues from using jwt initially? I cannot think of any really unless somehow they have xds with jwt auth disable but mtls enabled which would be odd

Reply:
We should do such an assumption that jwt will work. This may mask the real problem and increase the debugging difficulty

Reply:
Should I add a log? The current message is `cannot load key pair, using token instead` in citadel client – seems to point out why we're falling back to JWT pretty clearly.

Reply:
https://github.com/istio/istio/blob/master/security/pkg/nodeagent/cache/secretcache.go#L966

just want to mention that if we can use the cert to rotate cert(which means there is a cert under ProvCert **path**) we will always use that cert instead of token

the **token**  is designed only for bootstrapping at the first time. And **should not** be used in the cert rotation or refresh

Reply:
It seems like that initial bootstrap is broken at the moment.. 

The way this PR is implemented we use jwt the first request, then switch to certs. 
An alternative is delaying xDS until we have the certs.

It's honestly unclear to me why the certs don't seem to get fetched on the code path you linked to and then things just eventually work on the xds path.

Reply:
great that is the correct process: bootstrapping using the jwt at first request and then switch to certs

Reply:
A VM using JWT should not set PROV_CERT. 

PROV_CERT is intended for the cases where the user has a long-lived certificate that he is provisioning (using ansible, user-specific server/agent, etc ) - this flow will not use JWTs and will not call citadel either.

Reply:
after sync with steven the cause and the solution like this:

the root cause:  

1. xds proxy should use token for the first request rather than using the cert where at that time is not under the file path
2. sds agent depends on xds proxy to do the request successfully first to fetch the cert for future connection
3. after sds agent gets the cert ,the xds and sds both will use that cert for future grpc connection

solution: 
steven gives such logic here, if the cert not exists in the PROV_CERT path the getCertKeyPaths function will give an empty path which indicates the xds proxy to use the token instead of cert

@stevenctl  I suggest you append `for the first time bootstrapping` after the pr title